content/root/best-practice-guide.textile

```diff
@@ -61,7 +61,7 @@ The main factors are summarized here:
 * If you want to publish at very high rates or with the lowest possible realtime latencies, "Ably's realtime library would be best as it can publish at higher rates and lower latencies than is possible with the REST library due to message pipelining":https://knowledge.ably.com/should-i-use-the-rest-or-realtime-library.
 * If you want to get one-time info about the presence set of a channel, you can use the "REST library":/rest.
 * If you want to set up realtime communication with IoT devices on platforms for which there isn't an Ably Client Library SDK available, you can use the "MQTT protocol adapter":/mqtt.
-* If you wish to migrate from PubNub or Pusher to Ably, you can use the relevant "protocol adapter":/adapters to "switch to Ably":https://knowledge.ably.com/can-you-help-ease-the-pain-of-migration-from-another-data-stream-network, and then gradually move over to using the Ably Client Library SDKs.
+* If you wish to migrate from PubNub or Pusher to Ably, you can use the relevant "protocol adapter":https://ably.com/protocols to "switch to Ably":https://knowledge.ably.com/can-you-help-ease-the-pain-of-migration-from-another-data-stream-network, and then gradually move over to using the Ably Client Library SDKs.
```

Comment:
```suggestion
* If you wish to migrate from PubNub or Pusher to Ably, you can use the relevant "protocol adapter":https://ably.com/protocols to "switch to Ably":https://knowledge.ably.com/can-you-help-ease-the-pain-of-migration-from-another-data-stream-network, and then gradually move over to Ably using the Ably Client Library SDKs.
```

Reply:
Can you "move over" after having already switched?

`...switch to Ably, and then gradually move over to Ably...`

Reply:
We could make this a little clearer. What's happening is you can use your existing clients written for Pusher/PubNub with Ably by using the protocol adapter. Those clients don't realize they are talking to Ably. Then, when you have dev resources, you can reimplement the clients using the Ably Client SDK so that they are talking directly to Ably and not going through the protocol adapter.

Reply:
Yes, definitely. This feels like it could quickly get off-piste from fixing a broken link though and should probably be addressed in the content review for this page.

Reply:
I read this as having a dangling "to" .. as in "move over to <what?> using the Ably Client SDKs". I see that I was wrong!

Perhaps: "If you wish to migrate from PubNub or Pusher to Ably, you can use the relevant protocol adapter to switch to the Ably platform without modifying your existing protocols and client libraries. You can then gradually rewrite your applications using the Ably Client Library SDKs" over time.

Reply:
Mark H please capture for clarification in another ticket and we can circle back to it. Thanks!

Reply:
Logged in [DOC-435](https://ably.atlassian.net/browse/DOC-435).